src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+      case None =>
+        import scala.collection.JavaConversions.asScalaIterator
+        truthReader.iterator.toIterator
+    }).filter(v => v.isSNP && v.isBiallelic)
+  }
+}
+
+private object AssemblyStatistics {
+  def apply(counter: NumericCounter[Long]): AssemblyStatistics = {
+    if (counter.isEmpty) return AssemblyStatistics(0, 0, 0)
+
+    val numBases = counter.map { case (length, count) => length * count }.sum
+    val fiftyPercent  = numBases / 2.0
+    val ninetyPercent = numBases * 0.9
+
+    // N50/N90
+    var n50: Option[Long] = None
+    var n90: Option[Long] = None
+    var i = 0
+    var blockLengthSum = 0d
+    val countAndLengths = counter.toSeq.sortBy(_._1)
+    while (i < countAndLengths.length && (n50.isEmpty || n90.isEmpty)) {
+      blockLengthSum += countAndLengths(i)._1 * countAndLengths(i)._2
+
+      if (n50.isEmpty && fiftyPercent <= blockLengthSum) {
+        n50 = Some(countAndLengths(i)._1)
+      }
+      if (n90.isEmpty && ninetyPercent <= blockLengthSum) {
+        n90 = Some(countAndLengths(i)._1)
+      }
+
+      i += 1
+    }
+
+    // L50
+    i = countAndLengths.length - 1
+    blockLengthSum = 0
+    var l50: Long = 0
+    while (0 <= i && blockLengthSum < fiftyPercent) {
+      blockLengthSum += countAndLengths(i)._1 * countAndLengths(i)._2
+      l50 += countAndLengths(i)._2
+      i -= 1
+    }
+
+    (n50, n90) match {
+      case (Some(a), Some(b)) => AssemblyStatistics(n50=a, n90=b, l50=l50)
+      case _ => unreachable("Could not compute n50/n90: " + counter.toList)
+    }
+  }
+}
+
+/**
+  * @param n50 the smallest block length such that the bases covered by all blocks this length and shorter are at least
+  * 50% of the # of bases covered by all blocks.
+  * @param n90 the smallest block length such that the bases covered by all blocks this length and shorter are at least
+  * 90% of the # of bases covered by all blocks.
+  * @param l50 the smallest number of blocks that whose sum produces the n50 value.
+  */
+private case class AssemblyStatistics(n50: Long, n90: Long, l50: Long)
+
+object PhaseBlockLengthMetric {
+  val MetricExtension = PathUtil.replaceExtension(Paths.get(AssessPhasingMetric.MetricExtension), ".block_lengths.txt").toString
+}
+
+/** Provides the number of phased blocks of a given length.
+  *
+  * @param is_truth true if the blocks are from the known/truth dataset, false otherwise
+  * @param length the length of the phased block
+  * @param count the number of phased blocks of the given length.
+  */
+case class PhaseBlockLengthMetric
+( is_truth: Boolean,
+  length: Long = 0,
+  count: Long = 0
+) extends Metric
+
+object PhasingUtils {
+  def getPhasingSetId(ctx: VariantContext): Int = {
+    ctx.getGenotype(0).getAttributeAsInt("PS", -1) // Integer.valueOf(ctx.getGenotype(0).getExtendedAttribute("PS", -1).toString)
+  }
+}
+
+private[vcf] class OverlapDetectorByContig(dict: SAMSequenceDictionary) {
```

Comment:
I don't understand the need for this.  OverlapDetector already partitions internally by Contig, and I can't figure out what this adds.